src/lib/CRSF/CRSF.cpp

```diff
@@ -359,6 +362,23 @@ void ICACHE_RAM_ATTR CRSF::sendSyncPacketToTX(void *pvParameters) // in values i
 #endif
 
 #if defined(PLATFORM_ESP8266) || defined(TARGET_R9M_RX) || defined(UNIT_TEST)
+        void ICACHE_RAM_ATTR CRSF::RXhandleUARTout()
+        {
+            noInterrupts();
+            uint8_t peekVal = SerialOutFIFO.peek(); // check if we have data in the output FIFO that needs to be written
+            if (peekVal > 0)
+            {
+                if (SerialOutFIFO.size() > (peekVal))
```

Comment:
If we got a bad value in peekVal, does this ever recover?

Reply:
It would empty the buffer and roll around depending what values snuck in. Although I don't think this should ever happen